sources/parser/parser_strip_quotes.c

```diff
@@ -42,24 +25,18 @@ char	*new_strip_quotes(char *arg)
 		if (is_quote(arg[i]))
 		{
 			quote_type = is_quote(arg[i]);
-			new_arg = ft_str_remove_char(arg, i, quote_type);
-			if (new_arg == NULL)
-				return (NULL);
-			arg = new_arg;
-			i--;
-			i = next_quote_char(arg, i, quote_type);
-			if (is_quote(arg[i]) == quote_type)
+			i++;
+			while (arg[i] != quote_type)
 			{
-				new_arg = ft_str_remove_char(arg, i, quote_type);
-				if (new_arg == NULL)
-					return (NULL);
-				free(arg);
-				arg = new_arg;
+				if (!ft_vec_push(vec, arg[i]))
+					return (show_error(E_MALLOC, shell, "", 1));
+				i++;
 			}
-			i--;
-			len = ft_strlen(arg);
 		}
+		else
+			if (!ft_vec_push(vec, arg[i]))
```

Comment:
I remember the agreement to not use !. But to use == NULL/0. Just for consistency.